src/Microsoft.OData.Core/ODataInnerError.cs

```diff
@@ -30,17 +32,44 @@ public ODataInnerError()
         /// <summary>Initializes a new instance of the <see cref="T:Microsoft.OData.ODataInnerError" /> class with exception object.</summary>
         /// <param name="exception">The <see cref="T:System.Exception" /> used to create the inner error.</param>
         public ODataInnerError(Exception exception)
+            : this(exception, "internalexception")
         {
-            ExceptionUtils.CheckArgumentNotNull(exception, "exception");
 
-            this.Message = exception.Message ?? string.Empty;
-            this.TypeName = exception.GetType().FullName;
-            this.StackTrace = exception.StackTrace;
+        }
+
+        public ODataInnerError(Exception exception, string nestedObjectName)

```

Comment:
>nestedObjectName [](start = 59, length = 16)

if this is not allowed to be null, add a CheckArgumentNotNull; otherwise call out in XML Comments (once added) that it may be null.